### PR TITLE
[doc] E: Add build and run process

### LINKED ISSUE #308: [doc] Update setup, build and run docs
## Description
We need to update the Nanvix docs, updating the setup and adding microvm build and run.

### PR DESCRIPTION
**Pull request by rafael0121**
## Description

Update setup and add instructions to run Nanvix with Microvm and QEMU.
This closes #308

### CODE REVIEW

File: doc/setup.md [hunk 1]
```diff
@@ -1,46 +1,122 @@
...
+
+This section instructs you on how to compile and run Nanvix. 
+
+## Using Microvm
+
+Compile Microvm:
+
+```bash
+cd $WORKDIR/nanvix/src/microvm
+make all
```
Comment (ppenna) [inline doc/setup.md]: This should be supported from root dir, if not, we need to bump the pointer for the build repository to the latest one.

Comment (rafael0121) [reply doc/setup.md]: Yes, it is. The Kernel too, but i chose this way to show that is possible.

Comment (ppenna) [reply doc/setup.md]: We should prefer building from the root directory, as build artifacts will be placed on the same output directory (ie `bin/`) as other build artifacts (ie, `kernel`).

If one builds components individually, build artifacts will be scattered across the repository. As a result, one could even end up with two binaries for `microm` (one in `bin/` and another in` src/microvm/bin`).

Could you please update instructions to guide people to *aways* build from the root directory?

Comment (rafael0121) [reply doc/setup.md]: Sure. I got it.

**Review by ppenna — APPROVED** [review]

**Pull request merged**